### PR TITLE
Add Conv2dLayer/Conv3dLayer to fix PyTorch 2.9.1 CuDNN Conv3d bug

### PR DESCRIPTION
**Pull request by yhyang201**
## Motivation

Very thanks for the benchmark data in this PR: https://github.com/sgl-project/sglang/pull/19788

  - Add `Conv2dLayer`/`Conv3dLayer` in `sglang/srt/layers/conv.py`. Conv3dLayer enables unfold+linear by default to avoid the PyTorch 2.9.1 + CuDNN < 9.15 Conv3d   
  bug (pytorch/pytorch#168167). Conv2dLayer is a drop-in replacement for `nn.Conv2d` with linear optimization opt-in.
  - Migrate 3 Conv3d models (qwen3_vl, qwen2_vl, glm4v) and 12 Conv2d patch embedding models to use the new layers.
  - Remove `check_torch_2_9_1_cudnn_compatibility()` from `server_args.py`.
 
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by mickqian — APPROVED** [review]
we need some performance and accuracy report for this one
